src/main/scala/firrtl/passes/AnnotateMemMacros.scala

```diff
@@ -0,0 +1,167 @@
+package firrtl.passes
+
+import scala.collection.mutable.{ArrayBuffer,HashMap}
+import AnalysisUtils._
+import firrtl.WrappedExpression._
+import firrtl.ir._
+import firrtl._
+import firrtl.Utils._
+import firrtl.Mappers._
+
+case class AppendableInfo(fields: Map[String,Any]) extends Info {
+  def append(a: Map[String,Any]) = this.copy(fields = fields ++ a)
+  def append(a: (String,Any)): AppendableInfo = append(Map(a))
+  def get(f: String) = fields.get(f)
+  override def equals(b: Any) = b match {
+    case i: AppendableInfo => fields - "info" == i.fields - "info"
+    case _ => false
+  }
+}
+
+object AnalysisUtils {
+
+  def getConnects(m: Module) = {
+    val connects = HashMap[String, Expression]()
+    def getConnects(s: Statement): Unit = s match {
+      case s: Connect =>
+        connects(s.loc.serialize) = s.expr
+      case s: DefNode =>
+        connects(s.name) = s.value
+      case s: Block =>
+        s.stmts foreach getConnects
+      case _ =>
+    }
+    getConnects(m.body)
+    connects.toMap
+  }  
+
+  // only works in a module
+  def getConnectOrigin(connects: Map[String, Expression], node: String): Expression = {
+    if (connects contains node) getConnectOrigin(connects,connects(node))
+    else EmptyExpression
+  }
+
+  def checkLit(e: Expression) = e match {
+    case l : Literal => true
+    case _ => false
+  }
```

Comment:
I don't think so since you can match on Literal as below. You can also call .isInstanceOf[Literal] on anything which is only marginally more characters than checkLit